sockets/v0/Socket.h

```diff
@@ -116,9 +128,12 @@ class Socket {
      * @return SOCKET_ERROR_NONE on success, or an error code on failure
      */
     virtual socket_error_t recv(void * buf, size_t *len);
+
     /**
      * Receive a message with the sender address and port
      * This API is not valid for SOCK_STREAM
+     * If the socket has not been opened, returns SOCKET_ERROR_NULL_PTR
```

Comment:
I just followed the examples that were already there as consistency is important and following a pre-existing consistent style should be done. However I agree that we should probably use @retval for all instances of this type.